packages/fuel-indexer/src/executor.rs

```diff
@@ -126,6 +127,14 @@ pub fn run_executor<T: 'static + Executor + Send + Sync>(
         let mut num_empty_block_reqs = 0;
 
         loop {
+            if kill_switch.load(Ordering::SeqCst) {
+                info!("Kill switch flipped, stopping Indexer({indexer_uid}). <('.')>");
+                if kill_confirm.send(()).is_err() {
+                    error!("Unable to notifty listeners that Indexer({indexer_uid}) has stopped.");
+                };
```

Comment:
In the changelog, it's stated that we wait until the old indexer stops before replacing the indexer. Does this actually cause the service to wait until it's done? It appears that this would just send a message on the oneshot channel after the logging before exiting the executor loop; wouldn't it still be possible for the confirmation to have been sent but the WASM execution to continue to save data to the database?